pkg/util/machineset/util_test.go

```diff
@@ -0,0 +1,158 @@
+package util
+
+import (
+	"maps"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	annotationsutil "github.com/openshift/machine-api-operator/pkg/util/machineset/util.go"
+)
+
+func TestSettingAnnotations(t *testing.T) {
+	tests := []struct {
+		name                string
+		expectedAnnotations map[string]string
+		suppliedAnnotations map[string]string
+	}{
+		{
+			name: "Set adds CpuKey annotation",
+			suppliedAnnotations: map[string]string{
+				annotationsutil.CpuKey: "",
+			},
+			expectedAnnotations: map[string]string{
+				annotationsutil.CpuKey: "1",
+			},
+		},
+		{
+			name: "Set adds MemoryKey annotation",
+			suppliedAnnotations: map[string]string{
+				annotationsutil.MemoryKey: "",
+			},
+			expectedAnnotations: map[string]string{
+				annotationsutil.MemoryKey: "4Gi",
+			},
+		},
+		{
+			name: "Set adds GpuCount and GpuType annotations",
+			suppliedAnnotations: map[string]string{
+				annotationsutil.GpuCountKey: "",
+				annotationsutil.GpuTypeKey:  "",
+			},
+			expectedAnnotations: map[string]string{
+				annotationsutil.GpuCountKey: "1",
+				annotationsutil.GpuTypeKey:  "nvidia.com/gpu",
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			target := NewWithT(t)
+			target.SetAnnotations(tc.expectedAnnotations)
+			err := target.UpdateScaleFromZeroAnnotations()
+			if err != nil {
+				t.Errorf("Unexpected error updating ScaleFromZero annotations :%v", err)
+			}
+
+			observed := target.GetAnnotations()
+
+			if !maps.Equal(observed, tc.expectedAnnotations) {
+				t.Errorf("SetAnnotations() returned %v, expected %v", observed, tc.expectedAnnotations)
+			}
+		})
+	}
+}
+
+func TestParsingAnnotations(t *testing.T) {
+	tests := []struct {
+		name                string
+		expectedAnnotations map[string]string
+		expectedReturn      bool
+	}{
+		{
+			name: "Tests that key and value exists in annotation",
+			expectedAnnotations: map[string]string{
+				annotationsutil.CpuKey: "1",
+			},
+			expectedReturn: false,
+		},
+		{
+			name: "Tests that key and value exists in annotation",
+			expectedAnnotations: map[string]string{
+				annotationsutil.CpuKey: "",
+			},
+			expectedReturn: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			target := NewWithT(t)
+			target.SetAnnotations(tc.expectedAnnotations)
+			err := target.UpdateScaleFromZeroAnnotations()
+			if err != nil {
```

Comment:
probably should remove this or the test will stop whenever we get a failure here